### PR TITLE
feat: make Cognito app client readable and writable attributes configurable

### PR DESCRIPTION
**Pull request by hallettj**

### CODE REVIEW

File: aws/cognito_user_pool/main.tf [hunk 1]
```diff
@@ -155,8 +155,8 @@ resource "aws_cognito_user_pool_client" "this" {
 
   generate_secret               = false
   prevent_user_existence_errors = "ENABLED"
-  read_attributes               = local.readable_user_attributes
-  write_attributes              = local.writable_user_attributes
+  read_attributes               = coalesce(each.value.read_attributes, local.default_readable_user_attributes)
+  write_attributes              = coalesce(each.value.write_attributes, local.default_writable_user_attributes)
```
Comment (chrizkim) [inline aws/cognito_user_pool/main.tf:159]: this works well, but just to point out another option if you think it's cleaner, Terraform added a `defaults` function to go with the `optional` type for this purpose, e.g.:
```
locals {
  ...
  default_readable_user_attributes = ...
  default_writable_user_attributes = ...
  
  clients = defaults(var.clients, {
    read_attributes = local.default_readable_user_attributes
    write_attributes = local.default_writable_user_attributes
  })
}

resource "aws_cognito_user_pool_client" "this" {
  for_each = local.clients
  ...
  read_attributes  = each.value.read_attributes
  write_attributes = each.value.write_attributes
}
```
either way works, so just up to personal preference

Comment (JoelAtDeluxe) [reply aws/cognito_user_pool/main.tf:159]: I was thinking the same thing, but I'm also fine with either solution.

**Review by JoelAtDeluxe — APPROVED** [review]

File: aws/cognito_user_pool/main.tf (hunk 1 above)
Comment (hallettj) [reply aws/cognito_user_pool/main.tf:159]: Unfortunately I'm running into some trouble. I think that the `defaults` function might not actually work that way. According to [this comment](https://github.com/hashicorp/terraform/issues/28406#issuecomment-821124349) it's not possible to assign a list as a default value?

Comment (chrizkim) [reply aws/cognito_user_pool/main.tf:159]: heh, makes sense where they ran into trouble there. guess what you have will have to do

**Review by chrizkim — APPROVED** [review]

**Pull request merged**